README.md

```diff
@@ -118,3 +113,210 @@ This stack is IPv6-capable. Caveats:
+
+       mkdir certs
+       cd certs
+       for SUBNAME in desec www.desec get.desec checkip.dedyn checkipv4.dedyn checkipv6.dedyn dedyn www.dedyn update.dedyn update6.dedyn
+       do
+           ln -s cer ${SUBNAME}.${DOMAIN}.cer
+           ln -s key ${SUBNAME}.${DOMAIN}.key
+       done
+
+       cp ~/bin/certbot/config/live/${DOMAIN}/fullchain.pem cer
+       cp ~/bin/certbot/config/live/${DOMAIN}/privkey.pem key
+
+    The last two steps need to be repeated whenever the certificates are renewed.
+    While any location for the certificates is fine, the `certs/` folder is configured to be ignored by git so that private keys do not accidentally end up being committed.
+
+1. **Configure desec-stack.** As docker-compose application, desec-stack is configured by environment variables defined in the `.env` file in the project root directory.
+    Because it contains sensitive information for each deployment, `.env` is not part of the repository and ignored by git.
+    However, we ship `.env.default` and `.env.dev` with templates for production and development, respectively.
+    `.env.dev` is almost good enough for a basic development system, so let's use that as a basis:
+
+       sed "s/^DESECSTACK_DOMAIN=.*/DESECSTACK_DOMAIN=${DOMAIN}/" .env.dev > .env
+
+    Optionally, edit the file and
+    1. configure an email server host name, user name, and password to deliver emails can be included in `.env`. A convenient option is a MailTrap account.
+    2. adjust the network prefixes in `.env` to avoid collisions with other local networks.
+
+1. **Get desec-static.** Currently, a second clone is needed to start desec-stack. We are planning to remove this dependency.
+    Static is responsible for the static content (i.e. website) of desec-stack.
+    As it currently requires components which we may not distribute, the website in your deployment will be broken. This will not affect the API in any way.
+    In the project root,
+
+       rm static
+       git clone https://github.com/desec-io/desec-static.git static
+       mkdir static/ultima  # workaround for proprietary components
+
+1. **Run desec-stack.** To run desec-stack, use
+
+       ./dev
+
+    If you run desec-stack for the first time, this will require a couple of downloads and take a while.
+    Once it is up and running, you can query the API home endpoint:
+
+       http GET https://desec.${DOMAIN}/api/v1/
+
+    Congratulations, you have desec-stack up and running.
+    Of course, as this setup is only on your local machine, DNS information will not be published into the public DNS.
+    However, the desec-stack nameserver is available on localhost port 5321.
+    To check if desec-stack is working as expected, you can query the desec-stack nameserver locally for any information that you saved using your API.
+
+       EMAIL=john@example.com
+       PASSWORD=insecure
+       http POST https://desec.${DOMAIN}/api/v1/auth/users/ email:=\"${EMAIL}\" password:=\"${PASSWORD}\"
```

Comment:
This does not work if you don't set the email environment variables. Hm. Not sure what the best solution is?

Reply:
fixed by b75a0579fdea9cd168d8f2b0b028308841e0019b